docs/RFCs/0002-crashtracker-user-visable-log.md

```diff
@@ -0,0 +1,169 @@
+- `threads`:
+    This **optional** field contains an array of `Thread` objects.
+    In a multi-threaded program, the collector SHOULD collect the stacktraces of all active threads, and report them here.
+    A `Thread` object has the following fields:
+    - `crashed`: a boolean which tells if the thread crashed.
+    - `name`: Name of the thread (e.g. 'Thread 0').
+    - `stack`: The `StackTrace` of the thread
+    - `state`: **Optional**. Platform-specific state of the thread when its state was captured (CPU registers dump for iOS, thread state enum for Android, etc.).
+    See below for more details on how stacktraces are formatted.
+- `is_crash`:
+    Boolean true if the error was a crash, false otherwise.
+- `kind`:
+    The kind of error that occurred.
+    For a crash, a non-exhaustive set of options include "SigAbort", "SigBus", "SigSegv".
+    This field MAY be extended to include options such at "Panic", "UnhandledException" etc.
+- `message`:
+    A human readable string containing an error message associated with the stack trace.
+- `source_type`:
+    The string "crashtracking".
+- `stack`:
+    This represents the stack of the crashing thread.
+    See below for more details on how stacktraces are formatted.
+
+### Optional fields
+Any field not listed as "Required" is optional.
+In order to minimize logging overhead, producers SHOULD NOT emit anything for an optional field.
+Consumers MUST accept json with elided optional fields.
+
+- `counters`:
+    The crashtracker offers a mechanism for programs to register counters to track which operations were active at the time of the crash.
+    At present, this is only used by the profiler, but this may be extended in the future.
+- `files`:
+    The collector MAY collect useful files, such as `/proc/self/maps` or `/proc/meminfo`, and include them here.
+    Files are stored as a `Map<filename, contents>` where `contents` is an array of plain text strings, one per line.
+- `proc_info`: 
+    Currently, just tracks the PID of the crashing process.  
+    In the future, this may record additional info about the crashing process.
+- `siginfo`:
+    The name and signal number of the crashing signal (on UNIX systems)
+- `span_ids`: 
+    A vector of string identifiers, representing the active span ids at the time of program crash.
+    The collector SHOULD collect as many as it can, but MAY cap the number of spans that it tracks.
+- `trace_ids:`
+    A vector of string identifiers, representing the active span ids at the time of program crash.
+    The collector SHOULD collect as many as it can, but MAY cap the number of spans that it tracks.
+
+### Extensibility
+Future versions of the crashtracker MAY add additional fields.
+Parsers MUST accept unexpected optional fields, either by ignoring them, or by displaying them as additional data.
+The version number SHOULD be incremented for important optional fields, and MUST be incremented when a required field is added or removed.
```

Comment:
What is an 'important optional field'? This sounds rather vaguely and can be misinterpreted, I am afraid.

What about using semver with minor bump for changes to optional fields and major bump for changes in required fields? 
The reasoning is that a receiver would be able to process any data with the same major version because only optional fields are affected and the compatibility is not broken by any change to the required fields.

Reply:
changed to semver